src/client/Microsoft.Identity.Client/AuthScheme/PoP/InMemoryCryptoProvider.cs

```diff
@@ -24,7 +24,7 @@ public InMemoryCryptoProvider()
 
         public string CannonicalPublicKeyJwk { get; private set; }
 
-        public string CryptographicAlgorithm { get => "RS256"; }
+        public string CryptographicAlgorithm { get => "PS256"; }
```

Comment:
Is this true for both netframework and otherwise?

Reply:
Would really need @trwalke or @bgavrilMS to look at this one and comment

Reply:
I think ComputeCanonicalJwk needs to be updated as well.

Reply:
let me update and add a test for it as well

Reply:
@bgavrilMS can you please check to make sure if the update is correct? if not can you please suggest the correct update for ComputeCanonicalJwk  with PS256

Reply:
Yeah, this looks right to me according to the spec

Reply:
actually, i dont think we need to add it to the conical JWK because it is not a required member for the computation. This is probably why we didn't have "alg":"RSA256" to begin with. Since PS256 is essentially RS256 with PSS padding, I would imagine the required members for the JWK public key would be the same but not entirely sure. I doudt we need to add the "alg":"PS256" though

![image](https://github.com/user-attachments/assets/a0a22eb9-6a95-4f45-bca5-64f6de9658f0)

Reply:
checking with ESTS

Reply:
@trwalke per ESTS "Algorithm is not part of JWK, same key can be used to produce multiple different signature, like here RS256 or PS256"

Reply:
`alg` parameter is not included, as it is not necessary for the canonical representation used for computing the thumbprint

By including only the required parameters (e, kty, and n) in lexicographic order, this method ensures that the JWK is correctly canonicalized for thumbprint computation as specified in RFC 7638.